pandas/_libs/hashtable_func_helper.pxi.in

```diff
@@ -169,15 +185,27 @@ cdef duplicated_{{dtype}}(const {{dtype}}_t[:] values, object keep='first'):
         with nogil:
         {{endif}}
             for i in range(n):
-                value = {{to_c_type}}(values[i])
-                k = kh_get_{{ttype}}(table, value)
-                if k != table.n_buckets:
-                    out[table.vals[k]] = 1
-                    out[i] = 1
+                if uses_mask and mask[i]:
+                    if not seen_na:
```

Comment:
So this `if/elif` block is when `keep=False`? I am having troubling understanding why we need to track first 2 NAs instead of just assigning `out[i] = 1`

Reply:
Yes correct. When keep=False we can only set True when we have multiple nas, so we have to set the first na to true after we saw a second one

Reply:
Oh duh, right. Need at least 2 NAs to be a duplicate 🤦